assets/shared/helpers/player.js

```diff
@@ -0,0 +1,552 @@
+const VIDEO_TYPE = 'video';
+const VIDEOPRESS_TYPE = 'videopress';
+const YOUTUBE_TYPE = 'youtube';
+const VIMEO_TYPE = 'vimeo';
+
+const players = {
+	[ VIDEO_TYPE ]: {
+		/**
+		 * Initialize the player.
+		 *
+		 * @param {HTMLVideoElement} element The player element.
+		 *
+		 * @return {Promise<HTMLVideoElement>} The video player through a promise.
+		 */
+		initializePlayer: ( element ) => {
+			// Return that it's ready when it can get the video duration.
+			if ( ! isNaN( element.duration ) ) {
+				return Promise.resolve( element );
+			}
+
+			return new Promise( ( resolve ) => {
+				element.addEventListener( 'durationchange', () => {
+					resolve( element );
+				} );
+			} );
+		},
+
+		/**
+		 * Get the video duration.
+		 *
+		 * @param {HTMLVideoElement} player The player element.
+		 *
+		 * @return {Promise<number>} The duration of the video in seconds through a promise.
+		 */
+		getDuration: ( player ) => Promise.resolve( player.duration ),
+
+		/**
+		 * Set the video to a current time.
+		 *
+		 * @param {HTMLVideoElement} player  The player element.
+		 * @param {number}           seconds The video time in seconds to set.
+		 *
+		 * @return {Promise} A promise that resolves if the video was set to a current time successfully.
+		 */
+		setCurrentTime: ( player, seconds ) => {
+			try {
+				player.currentTime = seconds;
+				return Promise.resolve();
+			} catch ( e ) {
+				return Promise.reject( e );
+			}
+		},
+
+		/**
+		 * Play the video.
+		 *
+		 * @param {HTMLVideoElement} player The player element.
+		 *
+		 * @return {Promise} The native promise from the video play function.
+		 */
+		play: ( player ) => player.play(),
+
+		/**
+		 * Pause the video.
+		 *
+		 * @param {HTMLVideoElement} player The player element.
+		 *
+		 * @return {Promise} A promise that resolves if the video was paused successfully.
+		 */
+		pause: ( player ) => {
+			try {
+				player.pause();
```

Comment:
[Documentation](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/pause) is quite lazy regarding this `pause()` method. Is this pausing immediately and there is no way it could fail?

I am wondering if we should check for `.paused` [property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/paused) to resolve the promise.

Reply:
I like the idea to check if it was paused to resolve this `Promise`. https://github.com/Automattic/sensei/pull/5364/commits/fc5e2dbfc854ae68d38236da71351636467ceab2